File: PASCAL/Micro11.pas
program micro11;
var
    numero : integer;
    x : integer;

function Verifica(n :integer) : integer;
var
    res : integer;
begin
    if (n > 0) then
        res := 1
    else
    begin
        if(n < 0) then
            res := -1
        else
            res := 0;
    end;
    Verifica := res;
end;

begin
    write('Digite um numero: ');
    readln(numero);
    x := Verifica(numero);
    if (x = 1) then
        writeln('Numero positivo')
    else
        begin
            if(x = 0) then
                writeln('Zero')
            else
                writeln('Numero negativo');
        end;
end.

File: IR LLVM/Micro11/Micro11.c
#include <stdio.h>
#include <stdlib.h>

int verifica(int n){
    int res;

    if(n > 0)
        res = 1;
    if(n < 0)
        res = -1;
    if(n == 0)
        res = 0;
    return res;
}

int main(){
    int num;
    int res;
    printf("Digite um numero: ");
    scanf("%d", &num);
    res = verifica(num);
    switch(res){
        case -1:
            printf("Numero negativo\n");
            break;
        case 0:
            printf("Numero zero\n");
            break;
        case 1:
            printf("Numero positivo\n");
            break;
    }
    return 0;
}